### PR TITLE
arch-riscv: Add v*.m inst

### PR DESCRIPTION
**Pull request by FanYang98**
Add  vfirst.m vcpop.m vmsbf.m vmsif.m vmsof.m viota.m insts

All insts pass AtomicSimpleCPU,TimingSimpleCPU,MinorCPU,O3CPU internel tests.

Change-Id: I66680b7a25ebe53762eebefacf7c1ce334a5d52b

### CODE REVIEW

**Comment by FanYang98** [comment]
@ksco @jerinjoy @huxuan0307  
Please review and let me know if any changes are required.

File: src/arch/riscv/utility.hh [hunk 1]
```diff
@@ -224,6 +224,14 @@ elem_mask(const uint8_t* v0, const int index)
...
 }
 
+template<typename T>
+inline int
+vs_mask(T* vs, int index){
+    int idx = index / (sizeof(T)*8);
+    int pos = index % (sizeof(T)*8);
+    return (vs[idx] >> pos) & 1;
+}
+
```
Comment (huxuan0307) [inline src/arch/riscv/utility.hh]: I think template function is more flexible than `elem_mask`. It prevent us from calculation of size of different elements. Maybe we should update function `elem_mask` with template one, and add some assertion, such as:
```C++
static_assert(std::is_integral_v<T>);
```

```C++
- inline int
- elem_mask(const uint8_t* v0, const int index)
- {
-     int idx = index / 8;
-     int pos = index % 8;
-     return (v0[idx] >> pos) & 1;
- }
+ template<typename T>
+ inline int
+ elem_mask(T* vs, int index){
+    int idx = index / (sizeof(T)*8);
+    int pos = index % (sizeof(T)*8);
+    return (vs[idx] >> pos) & 1;
+ }

```

Comment (FanYang98) [reply src/arch/riscv/utility.hh]: I strongly agree with you, but I noticed that many instructions (50+) use `elem_mask` in `decoder.isa`. Using `elem_mask` template will cause a lot of modifications. Maybe we can accomplish this task in the other commit. And I think the assertion you mentioned is very helpful.

**Comment by FanYang98** [comment]
I've added some new commits to use template `elem_mask`. Once this looks good we can merge it.

**Pull request merged**